### DISCUSSION TITLE
Invert selection of nodes

### DISCUSSION THREAD

**Discussion by DanielFlaum** [freeplane/freeplane#921] [Ideas]
It would be helpful, in large maps when I need to do something to all but a few nodes, to be able to select the few nodes I don't want to modify and then invert the selection to select the many nodes I do want to modify.

I think there are three separate invert commands that would be useful:

1. **Invert selection globally**. This causes all formerly unselected nodes to become selected and all formerly selected nodes to become unselected at all levels of the map.
2. **Invert selection in branch**. This causes all formerly unselected siblings of the last selected node to become selected and all formerly selected siblings of the last selected node to become unselected.
3. **Invert selection at level**. This causes all formerly unselected nodes at the level of the last selected node to become selected and all formerly selected nodes at the level of the last selected node to become unselected.

The exact wording of the command names should be work-shopped, but we might find these commands already exist in other mind mapping, diagramming, or outlining software. They should be added to the Navigate menu, where there are already other select commands.

**Comment by sifran-github** [comment]
It might be problematic because usually on large maps some nodes are folded, and therefore cannot be selected.

Most of the time, if you want to change the format of all the nodes except a few, you should change the default style (format -> manage styles -> edit styles), and then just manually edit the style of the few nodes.

**Comment by dpolivaev** [comment]
To increase the chances of your idea being considered and potentially implemented, it's important to provide a detailed and convincing explanation of why your idea is necessary. This will help to clearly communicate the benefits and value of your suggestion, and may also lead to the generation of related ideas and cost-effective implementation solutions.

**Comment by macmarrum** [comment]
@DanielFlaum  I had a similar need and wrote some scripts. Please feel free to use them and report back.

- [invertSelectionGlobally.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/invertSelectionGlobally.groovy)
- [selectSiblings.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/selectSiblings.groovy)
- [invertSelectionAtLevel.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/invertSelectionAtLevel.groovy)

If you're new to the topic of scripts in Freeplane, please see https://docs.freeplane.org/scripting/Scripting.html

The scripts will appear in a new manu "Mac1". If you'd rather have them in Tools->Scripts, remove `="/menu_bar/Mac1/Select"` from each script (in the first line).
